### ISSUE TITLE
synth init fails creating config file on windows 10

### ISSUE THREAD

**Issue by JeyRathnam** [getsynth/synth#73]
**Describe the bug**
"synth init" fails to create config.toml file on windows 10. 

**To Reproduce**
1. run synth init on any folder in windows 10.
2. Error Message : Error: Failed to create config file at: \\?\C:\Users\jaira\Documents\Projects\synth\.github\workflows\scripts\.synth/config.toml during initialization  Caused by:
    The filename, directory name, or volume label syntax is incorrect. (os error 123)

**Expected behavior**
config.toml file should create under .synth directory.

**Screenshots**
![image](https://user-images.githubusercontent.com/6557230/127053014-c2cdfbb7-15d7-4528-9c2f-f49db7204928.png)

**Environment (please complete the following information):**
 - OS: windows 10 x64
 - Version: 0.5.3

**Additional context**
Researching about this, the issue is possibly due to "\\\\" at the start of the file path.

**Comment by fretz12** [comment]
Looks like some of the paths within synth has been hardcoded to use `/`, like you rightfully mentioned.
I'll look into making the paths OS-independent and make sure it runs properly on windows.

**Comment by christoshadjiaslanis** [comment]
Thanks @fretz12.

I would also expand the tests in `cli/mod.rs` to ->

```rust
#[cfg(test)]
pub mod tests {
    use super::*;
    use std::env::temp_dir;
    use crate::{META_OS, version};

    #[test]
    fn test_derive_seed() {
        assert_eq!(Cli::derive_seed(false, None).unwrap(), 0);
        assert_eq!(Cli::derive_seed(false, Some(5)).unwrap(), 5);
        assert!(Cli::derive_seed(true, Some(5)).is_err());
        assert!(Cli::derive_seed(true, None).is_ok());
    }

    #[test]
    fn test_init() {
        let args = CliArgs::Init { init_path: Some(temp_dir()) };
        let cli = Cli::new(args, version(), META_OS.to_string())
            .unwrap();
        assert!(cli.init(Some(temp_dir())).is_ok())
    }
}
```

Just to have a sanity check. I think this bug also makes it clear that we need to run the `synth-cargo-test` workflow on an OS matrix. We can do this here or as part of a different ticket. What do you think?

**Comment by fretz12** [comment]
@christoshadjiaslanis - good call, i'll include those test changes here, and agreed, let's keep the OS matrix testing enhancement here as I'll do that as part of the fix.

**Comment by llogiq** [comment]
Merged #77 manually, so this should now work. Please try and re-open if not.
